### ISSUE TITLE
Unable to obtain Dried Rose after save and reload

### ISSUE THREAD

**Issue by ansiyuan** [00-Evan/shattered-pixel-dungeon#58]
Suppose I complete the Sad Ghost's quest and save the game before the Dried Rose gets generated anywhere. After reloading, the probability of generating the Dried Rose is reset to 0 and I'll never be able to obtain it during the remainder of this game.

This issue can be fixed by saving artifact generation probabilities instead of a list of generated artifact names, or by checking the quest's status when restoring probabilities, which requires making `Ghost.Quest.processed` a `public` field (or providing an accessor method).
```java
public static void restoreFromBundle(Bundle bundle) {
	initArtifacts();
	if (Ghost.Quest.processed) {
		Category cat = Category.ARTIFACT;
		cat.probs[cat.classes.indexOf(DriedRose.class)] = 1;
	}
	if (bundle.contains(ARTIFACTS)) {
		// ...
	}
}
```

**Comment by 00-Evan** [comment]
Hmm, good catch.

The rose is somewhat underwhelming at the moment anyway, and I have some plans to make it a bit more fun, so I may let this bug hang around for a little bit so as people are more likely to get a better artifact, but definitely this should be fixed.

I'll admit that the choice of storing artifact class names is a bit questionable, I believe I did it because I wanted to be independent of the order of the list in the generator, should it ever be changed. Regardless changing it now only makes a larger mess as I would need to support old saves using the existent method for the forseeable future. Probably best is to just make `Ghost.Quest.processed` public and reference it.

**Comment by 00-Evan** [comment]
This issue has been addressed in the latest beta.

You can opt in to receive betas from google play here: https://play.google.com/apps/testing/com.shatteredpixel.shatteredpixeldungeon

Once the source is released this ticket will be closed with the commit referenced.

**Comment by 00-Evan** [comment]
Issue corrected in commit: 4220c22cbf510fe2fcd926c29ccccc51102ea9a0
